parsons/newmode/newmode_v2.py

```diff
@@ -0,0 +1,237 @@
+from parsons.utilities.oauth_api_connector import OAuth2APIConnector
+from parsons.utilities import check_env
+from parsons import Table
+import logging
+
+logger = logging.getLogger(__name__)
+
+API_URL = "https://base.newmode.net/api/"
+API_AUTH_URL = "https://base.newmode.net/oauth/token/"
+API_CAMPAIGNS_URL = "https://base.newmode.net/"
+
+
+class NewmodeV2(object):
+    def __init__(
+        self,
+        client_id=None,
+        client_secret=None,
+        api_version="v2.1",
+    ):
+        """
+        Instantiate Class
+        `Args`:
+            api_user: str
+                The username to use for the API requests. Not required if ``NEWMODE_API_URL``
+                env variable set.
+            api_password: str
+                The password to use for the API requests. Not required if ``NEWMODE_API_PASSWORD``
+                env variable set.
+            api_version: str
+                The api version to use. Defaults to v1.0
+        Returns:
+            NewMode Class
+        """
+        self.api_version = check_env.check("NEWMODE_API_VERSION", api_version)
+        self.base_url = API_URL
+        self.client_id = check_env.check("NEWMODE_API_CLIENT_ID", client_id)
+        self.__client_secret = check_env.check("NEWMODE_API_CLIENT_SECRET", client_secret)
+        self.headers = {"content-type": "application/json"}
+
+    def convert_to_table(self, data):
+        """Internal method to create a Parsons table from a data element."""
+        table = None
+        if type(data) is list:
+            table = Table(data)
+        else:
+            table = Table([data])
+
+        return table
+
+    def base_request(self, method, url, data=None, json=None, data_key=None, params={}):
+        response = None
+        response = self.client.request(
+            url=url, req_type=method, json=json, data=data, params=params
+        )
+        response.raise_for_status()
+        success_codes = [200, 201, 202, 204]
+        self.client.validate_response(response)
+        if response.status_code in success_codes:
+            if self.client.json_check(response):
+                response_json = response.json()
+                return response_json[data_key] if data_key else response_json
+            else:
+                return response.status_code
+        return response
+
+    def converted_request(
```

Comment:
This method needs a docstring, I'll wait for you to add it before asking questions about the method as I imagine some of my questions will be answered by the docstring